main.cpp

```diff
@@ -98,23 +99,58 @@ void button_callback(MbedCloudClientResource *resource, const NoticationDelivery
  * @param endpoint Information about the registered endpoint such as the name (so you can find it back in portal)
  */
 void registered(const ConnectorClientEndpointInfo *endpoint) {
-    printf("Connected to Pelion Device Management. Endpoint Name: %s\n", endpoint->internal_endpoint_name.c_str());
+    printf("Registered to Pelion Device Management. Endpoint Name: %s\n", endpoint->internal_endpoint_name.c_str());
 }
 
 int main(void) {
-    printf("Starting Simple Pelion Device Management Client example\n");
-    printf("Connecting to the network...\n");
-
-    // Connect to the internet (DHCP is expected to be on)
-    nsapi_error_t status = net->connect();
+    printf("\nStarting Simple Pelion Device Management Client example\n");
+
+#ifdef USE_BUTTON
+    // If the User button is pressed ons start, then format storage.
+    DigitalIn *user_button = new DigitalIn(BUTTON1);
+    const int PRESSED = 1;
+    if (user_button->read() == PRESSED) {
```

Comment:
We already have format code in Simple Cloud Client. Also why do we always need to format storage, this should never be necessary. Why can't this be fixed upstream.